### ISSUE TITLE
Sniff that detects incorrect class path/namespaces??

### ISSUE THREAD

**Issue by dmason30** [squizlabs/PHP_CodeSniffer#2163]
Is there a sniff that detects incorrect class path/namespaces (defined in PSR-4)?

     // path: src/app/Orders/Foo.php
    
    namespace App/Jobs; // incorrect namespace path should be App/Orders

    class Foo {
        ...
    }

**Comment by gsherwood** [comment]
There isn't an included one and I don't personally know of any custom standards that have one, but I have not gone looking.

**Comment by jrfnl** [comment]
Just wondering: @gsherwood would you be interested in including such a sniff in PHPCS ?

**Comment by gsherwood** [comment]
@jrfnl Yep

**Comment by Majkl578** [comment]
I think [this sniff](https://github.com/slevomat/coding-standard#slevomatcodingstandardfilestypenamematchesfilename) from Slevomat CS is what you're looking for.

**Comment by jrfnl** [comment]
@Majkl578 Interesting sniff and looks to be very configurable, though explicitly states that it's not for PSR-4. Does the docs need updating ?

> For projects not following the PSR-0 or PSR-4 autoloading standards

**Comment by Majkl578** [comment]
As I understand it you can use it for PSR-4 as well, but on top of that, it also supports non-PSR-4 project structures (by configuring multiple elements under rootNamespaces).

I'd suggest giving something like this a try:
```xml


<rule ref="SlevomatCodingStandard.Files.TypeNameMatchesFileName">
	<properties>
		<property name="rootNamespaces" type="array">
			<element key="app" value="App"/>
		</property>
	</properties>
</rule>
```
and see if it fits your needs.

**Comment by dmason30** [comment]
@Majkl578 Sorry, I forgot to reply. This worked for our needs. I will close this as there is a Sniff that exists so doesn't necessarily have to come with phpcs.
